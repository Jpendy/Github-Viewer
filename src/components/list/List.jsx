import React from 'react';
import PropTypes from 'prop-types';

const List = ({ repoList }) => {
    
  const listElements = repoList.map(item => {
    return (<li key={ item }>
      { item.name }
    </li>);
  });

  return (
    <ul>
      { listElements }
    </ul>
  );

};

List.propTypes = {
  repoList: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default List;
