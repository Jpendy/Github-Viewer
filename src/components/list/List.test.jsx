import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List repoList={[{ thing1: 'thing' }, { thing1: 'thing' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
