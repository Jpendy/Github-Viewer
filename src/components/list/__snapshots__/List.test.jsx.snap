// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul>
  <li
    key="[object Object]"
  >
    <a />
  </li>
  <li
    key="[object Object]"
  >
    <a />
  </li>
</ul>
`;
